pkg/stash/with_azureblob.go

```diff
@@ -37,15 +38,17 @@ func WithAzureBlobLock(conf *config.AzureBlobConfig, timeout time.Duration, chec
 		}
 	}
 	if conf.ManagedIdentityResourceID != "" {
-		spStorageToken, err := adal.NewServicePrincipalTokenFromManagedIdentity(conf.StorageResource, &adal.ManagedIdentityOptions{IdentityResourceID: conf.ManagedIdentityResourceID})
+		msiCred, err := azidentity.NewManagedIdentityCredential(&azidentity.ManagedIdentityCredentialOptions{
+			ID: azidentity.ResourceID(conf.ManagedIdentityResourceID),
+		})
 		if err != nil {
 			return nil, errors.E(op, err)
 		}
-		err = spStorageToken.Refresh()
+		token, err := msiCred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{conf.CredentialScope}})
 		if err != nil {
 			return nil, errors.E(op, err)
 		}
-		cred = azblob.NewTokenCredential(spStorageToken.OAuthToken(), nil)
+		cred = azblob.NewTokenCredential(token.Token, nil)
```

Comment:
please use something like:
	blobClient, err := azblob.NewClient(storageAccountUrl, cred, nil)
	please

Reply:
oh, since this sdk is old, you need to feed in the token refersh in the second parameter instead of nil. so the token can be refreshed automatically.

Reply:
yes, thank you for your good suggestion